pkg/kubelet/container/runtime_cache.go

```diff
@@ -80,6 +80,14 @@ func (r *runtimeCache) GetPods() ([]*Pod, error) {
 	return r.pods, nil
 }
 
+// getPods returns the current cached pods. This is used mainly for testing
+// because it does not start the refreshing thread.
+func (r *runtimeCache) getCachedPods() []*Pod {
+	r.Lock()
+	defer r.Unlock()
```

Comment:
yeah i get that, ill wait for the tests. it feels like if you're going to call this concurrently, just returning the list without checking times won't be useful and if you're calling it standalone you won't need the locks.